REGRESSION(273517@main): `findString:` skips every other match in PDFs

charliewolfe opened the pull request:
#### f506eadd86a56445d847dc8219aa3e5acb87ed70
<pre>
REGRESSION(273517@main): `findString:` skips every other match in PDFs
<a href="https://bugs.webkit.org/show_bug.cgi?id=269025">https://bugs.webkit.org/show_bug.cgi?id=269025</a>
<a href="https://rdar.apple.com/122588827">rdar://122588827</a>

Reviewed by Sammy Gill.

To prepare for site isolation, 273517@main made changes to have separate IPC messages for searching web
processes for a string and setting selection. To make this work correctly, I added `DoNotSetSelection`
to the `FindOptions` enum to indicate that a `FindString` message should be limited to search. However,
`PDFPlugin::findString()` did not respect this flag, so the selection would be updated twice for each
`FindString` message. Fix this by early returning before setting selection in `PDFPlugin::findString()`
when the `DoNotSetSelection` flag is set.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(WebKit::PDFPlugin::findString):

Canonical link: <a href="https://commits.webkit.org/274352@main">https://commits.webkit.org/274352@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e129f8d858b6b2ab2df7ea621097589e676dc603

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/38744 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/17675 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/41089 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/41275 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/34397 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/20599 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/15021 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/32513 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/39317 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/14944 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/33650 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/12928 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/12942 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/42551 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/35241 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/34911 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/38732 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/13554 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/11206 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/36942 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/15166 "Built successfully") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/14850 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5062 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/14647 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
```diff
@@ -1367,6 +1367,8 @@ static bool getEventTypeFromWebEvent(const WebEvent& event, NSEventType& eventTy
     // FIXME: How can always returning true without searching if passed a max of 0 be right?
     // Even if it is right, why not put that special case inside countFindMatches instead of here?
     bool foundMatch = !maxMatchCount || countFindMatches(target, options, maxMatchCount);
+    if (options.contains(WebCore::DoNotSetSelection))
```
Not in this PR, but mind making `FindOptionFlag` an `enum class FindOptionFlag : uint16_t` at some point?

aprotyas reviewed (commented):
LGTM

charliewolfe committed:
REGRESSION(273517@main): `findString:` skips every other match in PDFs
https://bugs.webkit.org/show_bug.cgi?id=269025
rdar://122588827

Reviewed by Sammy Gill.

To prepare for site isolation, 273517@main made changes to have separate IPC messages for searching web
processes for a string and setting selection. To make this work correctly, I added `DoNotSetSelection`
to the `FindOptions` enum to indicate that a `FindString` message should be limited to search. However,
`PDFPlugin::findString()` did not respect this flag, so the selection would be updated twice for each
`FindString` message. Fix this by early returning before setting selection in `PDFPlugin::findString()`
when the `DoNotSetSelection` flag is set.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(WebKit::PDFPlugin::findString):

Canonical link: https://commits.webkit.org/274352@main
